app/models/edition/timestamps.rb

```diff
@@ -0,0 +1,58 @@
+class Edition::Timestamps
+  def self.edited(
+    edition,
+    payload,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+    edition.temporary_first_published_at = previous_live_version&.first_published_at
```

Comment:
Since we have that `NoPreviousPublishedItem` class we could in theory accept that into this class and have that return nil.

As I'm typing this I'm wondering if that's a bad idea as it probably creates indirection - be harder still to work out why it's nil - but thought I'd leave this comment to see your thoughts. We end up a bit caught between fighting against a pattern in https://github.com/alphagov/publishing-api/pull/1007/files#diff-d07c485229519b45d396e1797345df7cR63 and having opportunity to continue it here

Reply:
At first, I felt like Edition::Timestamps shouldn't need to know about `NoPreviousPublishedItem` because it can't do much with it anyway e.g. `NoPreviousPublishedItem` will never have a `first_published_at` so what's the point in trying to call that on it.
Then, after reading the article about the Null Object Pattern, I can see the value in that.  I suppose my concern is having a bunch of methods that just return nil is a bit annoying.  We could alias them perhaps like this https://stackoverflow.com/questions/19979100/alias-a-method-multiple-times ... or is that really gross?

Reply:
I wouldn't be too worried about the bunch of methods. The whole object is a bunch of methods that return effectively null or default values. 

As they return nil I think you could define them as simply as `def first_published_at; end`